### ISSUE TITLE
the pre-trained weights of declip in EVA02-CLIP-L-14-336

### ISSUE THREAD

**Issue by yytang25** [xiaomoguhz/DeCLIP#5]
Thank you for this outstanding work.
Where can I download the pre-trained weights of declip in EVA02-CLIP-L-14-336?

**Comment by xiaomoguhz** [comment]
Sorry for the late reply. I have been busy with some new research work recently. We will provide the ViT-L version of DeCLIP as soon as possible.

**Comment by zxqing01** [comment]
I also look forward to your announcement of the pre-trained weights.

**Comment by yytang25** [comment]
> Sorry for the late reply. I have been busy with some new research work recently. We will provide the ViT-L version of DeCLIP as soon as possible.

I used the command `CUDA_VISIBLE_DEVICES=0 python eval.py --config configs/cfg_ade20k.py --work-dir logs/ade20k`, and the default `DeCLIP_evab_dinov2B_csa_560_0.25_seg` model, and the obtained indicators were **aAcc: 46.6100 mIoU: 20.0900 mAcc: 47.1900**. Is this indicator normal?
At the same time, I trained an `EVA02-CLIP-L-14` model myself, using the default configuration in config, but the indicators I got were **aAcc: 43.6800 mIoU: 20.4200 mAcc: 47.8700**, which is even worse. Why is this?

**Comment by xiaomoguhz** [comment]
Hello, @yytang25，may I ask which attention computation mode you used? As indicated by the name of our model, the mode should be set to 'csa'. If you use 'qq', you will get incorrect results. Finally, the results of this model on zero-shot semantic segmentation are as follows:
 | COCOstuff | VOC 20 | VOC 21 | ADE   | City  | COCO_Obj | Context 60 | Context 59 | AVG     |
|:---------:|:------:|:------:|:-----:|:-----:|:--------:|:----------:|:----------:|:-------:|
| 25.39     | 85.14  | 60.43  | 21.74 | 33.15 | 36.35    | 35.38      | 39.36      | 42.1175 | . As we mentioned in the CATSeg branch, due to an accidental operation in VSCode, we permanently deleted all the previous weights, so we had to retrain the model. You may notice that the performance of the model is somewhat different from what was reported in the paper, but the overall performance is similar.

**Comment by xiaomoguhz** [comment]
@yytang25 As we mentioned in the CATSeg branch, due to an accidental operation in VSCode, we permanently deleted all the previous weights, so we had to retrain the model. You may notice that the performance of the model is somewhat different from what was reported in the paper, but the overall performance is similar.

**Comment by yytang25** [comment]
> Hello, [@yytang25](https://github.com/yytang25)，may I ask which attention computation mode you used? As indicated by the name of our model, the mode should be set to 'csa'. If you use 'qq', you will get incorrect results. Finally, the results of this model on zero-shot semantic segmentation are as follows:
> COCOstuff 	VOC 20 	VOC 21 	ADE 	City 	COCO_Obj 	Context 60 	Context 59 	AVG
> 25.39 	85.14 	60.43 	21.74 	33.15 	36.35 	35.38 	39.36 	42.1175

When I trained the model, I used the default configuration of `dist_DeCLIP_eva_vitL14_336_coco.sh`. The mode used was `qq_vfm_distill`. I changed it to `csa_vfm_distill `and tried training again. Could you please help me check if the following configuration is correct now? Thank you!

<img width="1864" alt="Image" src="https://github.com/user-attachments/assets/ccfff136-396d-43c9-a6e2-7a69ad8e5a63" />

**Comment by xiaomoguhz** [comment]
@yytang25 Your settings are correct. Generally, for OV detection or segmentation tasks, we set the context weight lower, such as 0.05, which tends to improve region classification performance. For zero-shot semantic segmentation, as indicated by our model name `DeCLIP_evab_dinov2B_csa_560_0.25_seg`, we usually recommend setting `--loss_context_weight to 0.25.` Finally, in our experiments, DINOv2-L as a teacher model typically performs better in region classification, but its performance in zero-shot semantic segmentation is not as good as DINOv2-B.

**Comment by yytang25** [comment]
> [@yytang25](https://github.com/yytang25) Your settings are correct. Generally, for OV detection or segmentation tasks, we set the context weight lower, such as 0.05, which tends to improve region classification performance. For zero-shot semantic segmentation, as indicated by our model name `DeCLIP_evab_dinov2B_csa_560_0.25_seg`, we usually recommend setting `--loss_context_weight to 0.25.` Finally, in our experiments, DINOv2-L as a teacher model typically performs better in region classification, but its performance in zero-shot semantic segmentation is not as good as DINOv2-B.

Thank you for your detailed reply, I will give it a try

**Comment by YUCOCOCS** [comment]
Thank you for the outstanding contribution of this paper. I have successfully reproduced the results of DeCLIP based on the ViT-B/16 framework for open-vocabulary segmentation, and I look forward to the release of the pre-trained model for the ViT-L version of DeCLIP.

**Comment by Justlovesmile** [comment]
> Thank you for the outstanding contribution of this paper. I have successfully reproduced the results of DeCLIP based on the ViT-B/16 framework for open-vocabulary segmentation, and I look forward to the release of the pre-trained model for the ViT-L version of DeCLIP.

Hello, @YUCOCOCS. May I ask how you reproduced the DeCLIP-CATSeg results for open-vocabulary segmentation?
I’ve been trying to do the same using the [DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0](https://huggingface.co/xiaomoguhzz/DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0) weight, but the performance on my side is quite poor. When I directly use the pre-trained [DeCLIP-CATSeg weight](https://huggingface.co/xiaomoguhzz/CATSeg_DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0) provided by the author to evalutate, the performance is very close to what’s reported in the paper — which is great!

Did you make any specific modifications to the config or training setup (e.g., learning rate, number of epochs, loss weights, etc.)? It would be really helpful if you could share some details.

Here’s what I observed:
```bash
##############################################################################
# Using CLIP_PRETRAINED: "EVA02-CLIP-B-16", CACHE_DIR: "DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0.pt"#
##############################################################################
[06/29 05:56:38] d2.engine.defaults INFO: Evaluation results for ade20k_150_test_sem_seg in csv format:
[06/29 05:56:38] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 05:56:38] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 05:56:38] d2.evaluation.testing INFO: copypaste: 1.8325,14.9668,3.0550,21.2411
[06/29 06:00:46] d2.engine.defaults INFO: Evaluation results for ade20k_full_sem_seg_freq_val_all in csv format:
[06/29 06:00:46] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:00:46] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:00:46] d2.evaluation.testing INFO: copypaste: 1.2392,14.4881,2.9301,18.9001
[06/29 06:02:09] d2.engine.defaults INFO: Evaluation results for voc_2012_test_sem_seg in csv format:
[06/29 06:02:09] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:02:09] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:02:09] d2.evaluation.testing INFO: copypaste: 1.0006,3.9614,5.0058,19.8982
[06/29 06:03:47] d2.engine.defaults INFO: Evaluation results for voc_2012_test_background_sem_seg in csv format:
[06/29 06:03:47] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:03:47] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:03:47] d2.evaluation.testing INFO: copypaste: 3.4641,46.1570,7.6369,53.0399
[06/29 06:08:08] d2.engine.defaults INFO: Evaluation results for context_59_test_sem_seg in csv format:
[06/29 06:08:08] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:08:08] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:08:08] d2.evaluation.testing INFO: copypaste: 2.5464,7.9555,5.1720,17.4202
[06/29 06:17:05] d2.engine.defaults INFO: Evaluation results for context_459_test_sem_seg in csv format:
[06/29 06:17:05] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:17:05] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:17:05] d2.evaluation.testing INFO: copypaste: 0.9356,7.0879,1.8343,14.5989
```
